lib/spec/router.sw-spec.js

```diff
@@ -206,4 +206,116 @@ describe('Router instances', function () {
+      });
+
+      it('infix placeholders (/foo/*/bar etc.) should match arbitrary path chunks', function() {
+        var mw2 = function () {};
+
+        router.get('/foo/*', mw);
+        router.get('/foo/bar/*', mw2);
+
+        expect(router.match('get', '/')).to.be.empty;
+        expect(router.match('get', '/foo')).to.be.empty;
+        expect(router.match('get', '/foo/')).to.deep.equal([mw]);
+        expect(router.match('get', '/foo/bar')).to.deep.equal([mw]);
+        expect(router.match('get', '/foo/bar/baz')).to.deep.equal([mw, mw2]);
+      });
+
+      it('multiple placeholders (/foo/*/bar/* etc) should be allowed', function() {
+        var mw2 = function () {},
+            mw3 = function () {};
+
+        router.get('/pre/*/fix/*', mw);
+        router.get('*/suf/*/fix', mw2);
+        router.get('*inner*', mw3);
+
+        expect(router.match('get', '/')).to.be.empty;
+        expect(router.match('get', '/pre/123/fix/')).to.deep.equal([mw]);
+        expect(router.match('get', '/pre/123/fix/456')).to.deep.equal([mw]);
+        expect(router.match('get', '/pre/123/')).to.be.empty;
+
+        expect(router.match('get', '/matched/suf/123/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/any/other/suf/123/456/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/suf/nopre/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/fix')).to.be.empty;
+
+        expect(router.match('get', '/pre/suf/fix/')).to.deep.equal([mw]);
+
+        expect(router.match('get', '/inner')).to.deep.equal([mw3]);
+        expect(router.match('get', '/beginners')).to.deep.equal([mw3]);
+        expect(router.match('get', '/pre/inner/fix/')).to.deep.equal([mw, mw3]);
+        expect(router.match('get', '/inner/suf/other/fix/')).to.deep.equal([mw2, mw3]);
+
+        expect(router.match('get', '/pre/123/fix/inner/suf/456/fix/')).to.deep.equal([mw, mw2, mw3]);
+        expect(router.match('get', '/pre/suf/inner/fix/')).to.deep.equal([mw, mw2, mw3]);
+      });
+
+      it('named placeholders (:foo) should match as anonymous placeholders (*)', function() {
+        router.get('/:foo', mw);
+        expect(router.match('get', '/')).to.deep.equal([mw]);
+        expect(router.match('get', '/foo/bar')).to.deep.equal([mw]);
+      });
+
+      xit('placeholder crowding (**) should ??? ', function() {
+        router.get('/foo/**', mw);
+        router.get('/f**bar', mw);
+        //expect(router.match('get', '/')).to.be.?;
+        //throw? degrade gracefully? treat as single placeholder? other?
+      });
+
+    });
+
+    describe('_parseSimplePath()', function() {
+      xit('on invalid registrations it should ???', function() {
+        expect(router._parseSimplePath('][')).to.not.throw;
+        // not throw? gracefully degrade? other?
+      });
+
+      it('should support anonymous placeholders (*)', function() {
+        expect(router._parseSimplePath('/a*b').source).to.equal("\\/a(.*?)b");
```

Comment:
Try to stress this method a little bit more. For instance, what if I want an asterisk?

Should I use `('/a\*b')`?